prerendering.bs

```diff
@@ -429,7 +435,29 @@ Every {{Document}} has an <dfn for="Document">activation start time</dfn>, which
 
   <!-- End copied section. -->
 
-  1. TODO prepend the existing session history of |predecessorBC| into |successorBC|? Or, probably better, use the new "browsing session" concept to bridge them? Be sure to respect |historyHandling|.
+  1. <p class="XXX">Appropriately merge the session histories of |predecessorBC| and |successorBC|, respecting |historyHandling|.</p>
+
+     This step cannot be specified precisely given the current sorry state of [[HTML]]'s session history specification. Once the <a href="https://github.com/whatwg/html/pull/6315">navigation and session history rewrite</a> pull request is merged, we can make this precise (although we will also have to do several updates to other parts of this prerendering specification).
+
+     In the meantime, let's be clear about the intent of the eventual precise-spec, in order to ensure interoperability:
+
+     - |successorBC| and |predecessorBC| will still be separate browsing contexts. Before activation, they are each in separate <a href="https://whatpr.org/html/6315/history.html#traversable-navigable">traversable navigables</a> (a new concept introduced in the rewrite). Call those |successorTraversable| and |predecessorTraversable|. Both of these traversable navigables have their own session history list; |successorTraversable|'s will necessarily consist of a single session history entry.
+
+     - During activation, we will:
+
+       - Move |successorBC| from |successorTraversable| into |predecessorTraversable|.
+
+       - Move the session history entry |successorEntry| from |successorTraversable| into |predecessorTraversable|, as follows:
+
+          - If |historyHandling| is "`replace`", delete the currently-active session history entry of |predecessorTraversable|.
+
+          - If |historyHandling| is "`default`", delete every entry after the the currently-active session history entry of |predecessorTraversable|.
+
+          - Append |successorEntry| into the |predecessorTraversable|'s session history list, and make it the active session history entry. (As a side effect, this then makes |successorBC| the active browsing context of |predecessorTraversable|).
```

Comment:
Shouldn't this try to replace the current session history in place instead of append if  |historyHandling| is "replace"?

Reply:
If it's replace, we deleted the currently-active one, so appending at this point makes sense I think.

Reply:
I was thinking of the case when the currently-active one is not the last entry in the session history list (e.g. after a back navigation?) but maybe that's OK too?

Reply:
Oh, I think you're right, let me re-do it with that case in mind. Thank you!